web/utils_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build unit
+
+package web
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/control-center/serviced/zzk/registry"
+	"github.com/control-center/serviced/zzk/service"
+	. "gopkg.in/check.v1"
+)
+
+var (
+	privateIp string = "private-ip"
+	privatePort uint16 = 12345
+	hostIp  string = "host-ip"
+	hostPort uint16 = 22250
+	
+	serviceAddress string = fmt.Sprintf("%s:%d", privateIp, privatePort)
+	muxAddress string = fmt.Sprintf("%s:%d", hostIp, hostPort)	
+)
+
+type dialedServiceAddress struct{}
+func (e *dialedServiceAddress) Error() string {
+	return serviceAddress
+}
+
+type dialedMuxAddress struct{}
+func (e *dialedMuxAddress) Error() string {
+	return muxAddress
+}
+
+// Mock dialers that just return an error on dial based on the address given.
+type mockDialer struct{}
+func (d *mockDialer) Dial(network, address string) (net.Conn, error) {
+	// If the address is the serviceAddress, return a serviceDialerError,
+	// otherwise return a muxDialerError
+	if address == serviceAddress {
+		fmt.Printf("returning a serviceDialerError for address: %s\n", address)
+		return nil, &dialedServiceAddress{}
+	} else {
+		fmt.Printf("returning a muxDialerError for address: %s\n", address)
+		return nil, &dialedMuxAddress{}
+	}
+}
+
+func (s *TestWebSuite) TestDoNotMuxLocalConnections(c *C) {
+	// Make sure localhost is in our ip map. hostIp isn't a real
+	// ip, so adding/removing it from the map won't have any effect
+	// on other tests.
+	ipmap[hostIp] = struct{}{}
+	defer delete(ipmap, hostIp)
+
+	// Create an export to test with.
+	export := getMockExport()
+
+	_, err := getRemoteConnection(&export, &mockDialer{});
+	if err.Error() == muxAddress {
+		c.Errorf("getRemoteConnection tried to use the mux for a local address")
+	}
+}
+
+func (s *TestWebSuite) TestMuxRemoteConnections(c *C) {
+	// Create an export to test with.
+	export := getMockExport()
+
+	_, err := getRemoteConnection(&export, &mockDialer{});
+	if err.Error() == serviceAddress {
+		c.Errorf("getRemoteConnection used a direct connection for a remote address")
+	}
+}
+
+func getMockExport() registry.ExportDetails {
```

Comment:
I wouldn't use the term 'mock' here if you're going to use the stretchr mock framework.  
It returns a valid instance of  `registry.ExportDetails`